src/plots/Mesh/avtMeshPlot.C

```diff
@@ -347,21 +349,71 @@ avtMeshPlot::SetAtts(const AttributeGroup *a)
     atts = *(const MeshAttributes*)a;
 
     SetLineWidth(Int2LineWidth(atts.GetLineWidth()));
-    if (atts.GetMeshColorSource()==0)
+    if (atts.GetMeshColorSource() == MeshAttributes::Foreground)
     {
         SetMeshColor(fgColor);
     }
-    else
+    else if (atts.GetMeshColorSource() == MeshAttributes::MeshRandom)
+    {
+        unsigned char rgb[3] = {0,0,0};
+        unsigned char fg[3] = {static_cast<unsigned char>(fgColor[0]*255),
+                               static_cast<unsigned char>(fgColor[1]*255),
+                               static_cast<unsigned char>(fgColor[2]*255)};
+
+        //
+        // Try and get the color for the colorIndex'th color in the distinct
+        // discrete color table or, failing that, default discrete color table.
+        // NOTE: The index is mod'd by the number of control points.
+        //
+        avtColorTables *ct = avtColorTables::Instance();
+        int colorIndex = this->instanceIndex;
+        do
+        {
+            if (! ct->GetControlPointColor(ct->GetDefaultDiscreteColorTable(), colorIndex, rgb))
+                if (! ct->GetControlPointColor("distinct", colorIndex, rgb))
+                    break;
+            colorIndex++;
+        }
+        while ((colorIndex - this->instanceIndex) < 50 &&
+               (ct->PerceptualColorDistance(rgb, fg) <= ct->JustNoticeableColorDistance()));
```

Comment:
So this is cool. But I think you should compare against background color not foreground. I don't think we care  if the foreground color is randomly used, but we do care if the mesh disappears into the background.

Reply:
Oh, of course. Good catch!